### PR TITLE
Refactor `MongoDb`

### LINKED ISSUE #67: Separate the API from the database logic
Currently, the API endpoints access the MongoDb database directly.

### PR DESCRIPTION
**Pull request by grtlr**
Closes #67.

### CODE REVIEW

File: src/db/model/stardust/message.rs [hunk 1]
```diff
@@ -81,60 +70,238 @@ pub struct MessageMetadata {
...
+                None,
+            )
+            .await
     }
 
-    /// Returns the corresponding milestone index.
-    pub fn milestone_index(&self) -> u32 {
-        self.milestone_index
+    /// Aggregates outputs by transaction ids.
+    pub async fn outputs_by_transaction_id(
```
Comment (Alexandcoats) [inline src/db/model/stardust/message.rs:145]: `get_`?

File: src/db/model/stardust/message.rs [hunk 2]
```diff
@@ -81,60 +70,238 @@ pub struct MessageMetadata {
...
+            None,
+        )
+        .await
     }
 
-    /// Returns the inclusion state of the message.
-    pub fn inclusion_state(&self) -> &LedgerInclusionState {
-        &self.inclusion_state
+    /// Aggregates the transaction history for an address.
+    pub async fn transaction_history(
```
Comment (Alexandcoats) [inline src/db/model/stardust/message.rs:162]: `get_?`

File: src/db/model/stardust/mod.rs [hunk 3]
```diff
@@ -5,3 +5,10 @@
 pub mod message;
 /// Module containing the milestone model.
 pub mod milestone;
+
+/// These are the names of the collections that we create in the database. At some point we could use a macro to make
+/// this list nicer.
+mod collection {
+    pub const MESSAGE_RECORDS: &str = "stardust_message_records";
```
Comment (Alexandcoats) [inline src/db/model/stardust/mod.rs:12]: Any reason for the change from `_messages` and `_milestones` here?

File: src/db/model/status.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use crate::db::MongoDb;
+
+/// Provides the information about the status of the node.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Status {
+    network_name: String,
+}
+
+impl MongoDb {
+    /// Get the persistet [`Status`] from the database.
```
Comment (Alexandcoats) [inline src/db/model/status.rs:22]: typo

File: src/db/model/stardust/mod.rs (hunk 3 above)
Comment (grtlr) [reply src/db/model/stardust/mod.rs:12]: I wanted to be more explicit. For instance, the type that we store is `MessageRecord`, so I thought I'd bring these in line.

**Comment by Alexandcoats** [comment]
LGTM other than comments

File: src/db/model/stardust/message.rs [hunk 5]
```diff
@@ -81,60 +70,238 @@ pub struct MessageMetadata {
...
+                UpdateOptions::builder().upsert(true).build(),
+            )
+            .await
     }
 
-    /// Returns should reattach indicator.
-    pub fn should_reattach(&self) -> bool {
-        self.should_reattach
+    /// Aggregates the spending transactions
+    pub async fn spending_transaction(&self, transaction_id: String, idx: u16) -> Result<Option<Document>, Error> {
```
[the remark this answers is not in the document]
Comment (grtlr) [reply src/db/model/stardust/message.rs:116]: You're right, we should prefix non-trivial getters with `get_`. 👍

**Review by Alexandcoats — DISMISSED** [review]

**Review by Alex6323 — DISMISSED** [review]

**Pull request merged**